core/primitives/address/impl/address_builder_impl.cpp

```diff
@@ -21,14 +21,14 @@ namespace fc::primitives::address {
   outcome::result<Address> AddressBuilderImpl::makeFromSecp256k1PublicKey(
       Network network, const Sec256k1PublicKey &public_key) noexcept {
     OUTCOME_TRY(hash, blake2b_160(public_key));
-    std::vector<uint8_t> sec256k1_bytes{network, Protocol::SECP256K1};
```

Comment:
Can we avoid calling `decode`?
Something like `return Address{MAINNET, Secp256k1PublicKeyHash(hash)};`
Same for bls.

Also may add TODO for [FIL-118](https://soramitsu.atlassian.net/browse/FIL-118).

Reply:
Secp256k1 and BLS Addresses are constructed in different ways. Secp256k1 uses hash of public key when bls just copies it. In my opinion code of creation shouldn't duplicate existing code even if it simple.